### PR TITLE
removed includes of CL/sycl.hpp

### PR DESCRIPTION
**Pull request by will-saunders-ukaea**
# Description

SYCL2020 states the includes should be of `SYCL/sycl.hpp`. Apart from the original 1D code (which performs it's own checks of if SYCL/sycl.hpp exists) NESO includes sycl transitively from NP. This PR removes includes and calls to "using namespace cl" which refer to how sycl worked pre 2020.

## Type of change

Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change)

# Testing

Built and runs with hipsycl 0.9.4 and dpcpp 2021.

### CODE REVIEW

**Review by oparry-ukaea — COMMENTED** [review]
Builds with oneapi for me.
With gcc I get stuff like
```
In file included from /home/oparry/code/NESO/src/linear_interpolator_1D.cpp:1:
  >> 80     /home/oparry/code/NESO-1/include/linear_interpolator_1D.hpp:49:3: error: reference to 'sycl' is ambiguous
     81        49 |   sycl::buffer<double, 1> buffer_x_data;
     82           |   ^~~~
```
cmake-format config and associated changes to format_all.sh work for me.

File: include/sycl_typedefs.hpp [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#ifndef _NESO_SYCL_TYPEDEFS_H_
+#define _NESO_SYCL_TYPEDEFS_H_
+
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+using namespace cl;
```
[the remark this answers is not in the document]
Comment (will-saunders-ukaea) [reply include/sycl_typedefs.hpp:8]: It "should" always be the first case with sycl2020. I think the "reference to 'sycl' is ambiguous" warnings are from NP using the CL/sycl.hpp on this submodule commit. Is it just that one file you see the warnings?

[the remark this answers is not in the document]
Comment (will-saunders-ukaea) [reply include/sycl_typedefs.hpp:8]: I can reproduce this locally - had a slightly different version of NP checked out.

[the remark this answers is not in the document]
Comment (will-saunders-ukaea) [reply include/sycl_typedefs.hpp:8]: @oparry-ukaea Should now be fixed with commit 1361636.

[the remark this answers is not in the document]
Comment (oparry-ukaea) [reply include/sycl_typedefs.hpp:8]: Yep, building for me with gcc now too.

**Review by oparry-ukaea — APPROVED** [review]
All looks to me now. Shall I merge?

**Pull request merged**